tensorboard/compat/tensorflow_stub/pywrap_tensorflow.py

```diff
@@ -177,51 +183,54 @@ def read(self):
         if self.filename is None:
             raise errors.NotFoundError(
                 None, None, 'No filename provided, cannot read Events')
-        if not os.path.exists(self.filename):
+        if not gfile.exists(self.filename):
             raise errors.NotFoundError(
                 None, None,
                 '{} does not point to valid Events file'.format(self.filename))
 
         # TODO: Handle gzip and zlib compressed files
-        with open(self.filename, "rb") as f:
-            buf = f.read()
-            n = self.start_offset
+        with gfile.GFile(self.filename, 'rb') as f:
+            f.read(self.start_offset)
 
-            while n < len(buf):
+            while True:
                 # Read the header
-                header_str = buf[n:n+8]
+                header_str = f.read(8)
+                if len(header_str) != 8:
+                    break  # Hit EOF so exit
                 header = struct.unpack('Q', header_str)
-                n += 12
 
-                # Read the crc32, which is 4 bytes, and check it against the
-                # crc32 of the header
-                # crc_header_str = buf[n:n+4]
-                # crc_header = struct.unpack('I', crc_header_str)
-                # n += 4
-                # header_crc_calc = masked_crc32c(header_str)
-                # assert header_crc_calc == crc_header[0], \
-                #     'Header crc\'s dont match'
+                # Read the crc32, which is 4 bytes, and check it against
```

Comment:
Note that we now do the crc32 checks in all cases. If perf is an issue we consider changing in the future.